### PR TITLE
Decode unsigned token

### PR DESCRIPTION
**Pull request by balanza**
Should we decode unsigned token?

### CODE REVIEW

**Comment by grausof** [comment]
what do you need it for?

File: src/sign.ts [hunk 1]
```diff
@@ -177,4 +178,16 @@ export class SignJWT extends ProduceJWT {
...
+   /**
+   * Decodes a JWT without signature
+   *
+   * @param jwtWithoutSignature JWT to sign that needs to be decoded.
+   */
+  static encode(jwt: {
+    payload: JWTPayload;
+  header: JWSHeaderParameters;
+  }): string {
+    return [jwt.header, jwt.payload].map(e =>JSON.stringify(e)).map(encodeBase64).join(".")
```
Comment (grausof) [inline src/sign.ts]: is decode or encode? This looks to me like the code of an encode.

**Review by grausof — CHANGES_REQUESTED** [review]

File: src/sign.ts (hunk 1 above)
Comment (balanza) [reply src/sign.ts]: It's actually a typo, shouldn't be in this PR. I remove it

File: src/sign.ts [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import { JOSENotSupported, JWSInvalid, JWTInvalid } from './utils/errors';
 import type {
   CompactJWSHeaderParameters,
+  JWSHeaderParameters,
```
Comment (grausof) [inline src/sign.ts:4]: 'JWSHeaderParameters' is defined but never used

**Review by grausof — CHANGES_REQUESTED** [review]

**Comment by balanza** [comment]
> what do you need it for?

@grausof actually nothing, I faced this problem while playing with some mocked JWTs and I wondered if that could be useful and/or correct.

I digged into [JWTs spec](https://www.rfc-editor.org/rfc/rfc7519#section-6) and I found this:
> To support use cases in which the JWT content is secured by a means
   other than a signature and/or encryption contained within the JWT
   (such as a signature on a data structure containing the JWT), JWTs
   MAY also be created without a signature or encryption.  An Unsecured
   JWT is a JWS using the "alg" Header Parameter value "none" and with
   the empty string for its JWS Signature value, as defined in the [JWA](https://www.rfc-editor.org/rfc/rfc7519#ref-JWA)
   specification [JWA]; it is an Unsecured JWS with the JWT Claims Set
   as its JWS Payload.
   
So my conclusion is that _in theory_ it might be correct to assume that a JWT with only two parts it's still a valid token (nevertheless we should consider some sort of validation on the `alg` attribute to be `none`). However, I see no actual use of that.

**Comment by balanza** [comment]
> what do you need it for?

@grausof actually nothing, I faced this problem while playing with some mocked JWTs and I wondered if that could be useful and/or correct.

I digged into [JWTs spec](https://www.rfc-editor.org/rfc/rfc7519#section-6) and I found this:
> To support use cases in which the JWT content is secured by a means
   other than a signature and/or encryption contained within the JWT
   (such as a signature on a data structure containing the JWT), JWTs
   MAY also be created without a signature or encryption.  An Unsecured
   JWT is a JWS using the "alg" Header Parameter value "none" and with
   the empty string for its JWS Signature value, as defined in the [JWA](https://www.rfc-editor.org/rfc/rfc7519#ref-JWA)
   specification [JWA]; it is an Unsecured JWS with the JWT Claims Set
   as its JWS Payload.
   
So my conclusion is that _in theory_ it might be correct to assume that a JWT with only two parts it's still a valid token (nevertheless we should consider some sort of validation on the `alg` attribute to be `none`). However, I see no actual use of that.

So I close the PR.

**Comment by grausof** [comment]
> > what do you need it for?
> 
> @grausof actually nothing, I faced this problem while playing with some mocked JWTs and I wondered if that could be useful and/or correct.
> 
> I digged into [JWTs spec](https://www.rfc-editor.org/rfc/rfc7519#section-6) and I found this:
> 
> > To support use cases in which the JWT content is secured by a means
> > other than a signature and/or encryption contained within the JWT
> > (such as a signature on a data structure containing the JWT), JWTs
> > MAY also be created without a signature or encryption.  An Unsecured
> > JWT is a JWS using the "alg" Header Parameter value "none" and with
> > the empty string for its JWS Signature value, as defined in the [JWA](https://www.rfc-editor.org/rfc/rfc7519#ref-JWA)
> > specification [JWA]; it is an Unsecured JWS with the JWT Claims Set
> > as its JWS Payload.
> 
> So my conclusion is that _in theory_ it might be correct to assume that a JWT with only two parts it's still a valid token (nevertheless we should consider some sort of validation on the `alg` attribute to be `none`). However, I see no actual use of that.
> 
> So I close the PR.

Yep 😉
if you take a look at the code there is also the [unsecured jwt](https://github.com/pagopa/io-react-native-jwt/blob/main/src/unsecured.ts)

**Pull request closed without merging**